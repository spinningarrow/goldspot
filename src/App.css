.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

body {
	margin: 2rem auto;
	max-width: 600px;
}

.song-list ul {
	list-style: none;
	padding: 0;
}

.song-item {
	margin-top: 0.5rem;
}

.song-item img {
	display: inline-block;
	margin-right: 0.5rem;
	vertical-align: middle;
	width: 32px;
}

.song-item span {
	display: inline-block;
	vertical-align: middle;
}
