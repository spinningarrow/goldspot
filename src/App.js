import React, { useState } from 'react'

import DataFetcher from './DataFetcher'
import Header from './Header'
import MultiSongList from './MultiSongList'
import Player from './Player'
import SongList from './SongList'
import {
	getAudioFeatures,
	getNowPlaying,
	getRecentlyPlayed,
	getTracks,
} from './api'

const App = () => {
	const [audioFeaturesSongList, setAudioFeaturesSongList] = useState(
		'energize'
	)

	const [enabledFeatures, setEnabledFeatures] = useState([])

	return (
		<div className="app">
			{enabledFeatures.includes('Player') && (
				<DataFetcher
					data={getNowPlaying}
					render={({
						items: { artist, trackName, isPlaying } = {},
					}) => (
						<Player
							artist={artist}
							trackName={trackName}
							isPlaying={isPlaying}
						/>
					)}
				/>
			)}

			<Header secretAction={setEnabledFeatures} />

			<main>
				<DataFetcher
					data={getRecentlyPlayed}
					render={({ items }) => (
						<SongList heading="Recently Played" items={items} />
					)}
				/>

				<DataFetcher
					data={getTracks}
					render={({ items }) => (
						<SongList items={items} heading="Recently Added" />
					)}
				/>

				<DataFetcher
					data={getAudioFeatures}
					render={({ items }) => (
						<MultiSongList
							selectedSongList={audioFeaturesSongList}
							handleSelection={setAudioFeaturesSongList}
							items={items}
						/>
					)}
				/>
			</main>
		</div>
	)
}

export default App
