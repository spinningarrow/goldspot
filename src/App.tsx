import React, { useState } from 'react'
import { useQuery, ApolloProvider } from '@apollo/client'

import DataFetcher from './DataFetcher'
import Header from './Header'
import Library from './Library'
import MultiSongList from './MultiSongList'
import Player from './Player'
import RecentlyAdded from './RecentlyAdded'
import RecentlyPlayed from './RecentlyPlayed'
import { client } from './client'
import { getAudioFeatures } from './api'
import { currentViewQuery } from './graphql-api'

client.writeQuery({ query: currentViewQuery, data: { currentView: 'played' } })

const App = () => {
	const [audioFeaturesSongList, setAudioFeaturesSongList] = useState(
		'energize'
	)

	const [enabledFeatures, setEnabledFeatures] = useState([])
	const { data } = useQuery(currentViewQuery, { client })
	const { currentView } = data

	return (
		<ApolloProvider client={client}>
			<div className="app">
				<Player />

				<Header secretAction={setEnabledFeatures} />

				<main>
					{currentView === 'played' && <RecentlyPlayed />}

					{currentView === 'added' && <RecentlyAdded />}

					{currentView === 'library' && <Library />}

					{currentView === 'features' && (
						<DataFetcher
							data={getAudioFeatures}
							render={({ items }) => (
								<MultiSongList
									selectedSongList={audioFeaturesSongList}
									handleSelection={setAudioFeaturesSongList}
									items={items}
								/>
							)}
						/>
					)}
				</main>
			</div>
		</ApolloProvider>
	)
}

export default App
