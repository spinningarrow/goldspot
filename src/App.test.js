import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import App from './App'

const renderer = new ShallowRenderer()

test('it renders without crashing', () => {
	const tree = renderer.render(<App recentlyAdded={[]} recentlyPlayed={[]} />)
	expect(tree).toMatchSnapshot()
})
