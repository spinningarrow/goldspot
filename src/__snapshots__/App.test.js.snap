// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders without crashing 1`] = `
<div
  className="app"
>
  <Header
    secretAction={[Function]}
  />
  <main>
    <DataFetcher
      data={[Function]}
      render={[Function]}
    />
    <DataFetcher
      data={[Function]}
      render={[Function]}
    />
    <DataFetcher
      data={[Function]}
      render={[Function]}
    />
  </main>
</div>
`;
